index.js

```diff
@@ -0,0 +1,241 @@
+            if (config.scmContext == null || config.scmContext.length === 0) {
+                reject('Not implemented');
+            }
+
+            const scm = this.scms[config.scmContext];
+
+            if (typeof scm !== 'object') {
+                reject('Not implemented');
+            }
+
+            resolve(callback(scm));
+        });
+    }
+
+    /**
+     * Process by all scm module
+     * @method allScm
+     * @param  {Function}   callback            {Map} fn(scm)
+     * @return {Map}                            combined callback results
+     */
+    allScm(callback) {
+        let result = {};
+
+        return new Promise((resolve, reject) => {
+            async.eachSeries(Object.values(this.scms), (scm, cb) => {
+                callback(scm)
+                .then((ret) => {
+                    result = Object.assign(result, ret);
+                    cb(null);
+                }, err => cb(err));
+            }, (err) => {
+                if (err) {
+                    reject(err);
+                }
+
+                resolve(result);
+            });
+        });
+    }
+
+    _addWebhook(config) {
+        return this.chooseScm(config, scm => scm.addWebhook(config));
+    }
+
+    _parseUrl(config) {
+        return this.chooseScm(config, scm => scm.parseUrl(config));
+    }
+
+    _parseHook(headers, payload) {
+        return this.chooseWebhookScm(headers, payload, scm => scm.parseHook(headers, payload));
+    }
+
+    _getCheckoutCommand(config) {
+        return this.chooseScm(config, scm => scm.getCheckoutCommand(config));
+    }
+
+    _decorateUrl(config) {
+        return this.chooseScm(config, scm => scm.decorateUrl(config));
+    }
+
+    _decorateCommit(config) {
+        return this.chooseScm(config, scm => scm.decorateCommit(config));
+    }
+
+    _decorateAuthor(config) {
+        return this.chooseScm(config, scm => scm.decorateAuthor(config));
+    }
+
+    _getPermissions(config) {
+        return this.chooseScm(config, scm => scm.getPermissions(config));
+    }
+
+    _getCommitSha(config) {
+        return this.chooseScm(config, scm => scm.getCommitSha(config));
+    }
+
+    _updateCommitStatus(config) {
+        return this.chooseScm(config, scm => scm.updateCommitStatus(config));
+    }
+
+    _getFile(config) {
+        return this.chooseScm(config, scm => scm.getFile(config));
+    }
+
+    _getOpenedPRs(config) {
+        return this.chooseScm(config, scm => scm.getOpenedPRs(config));
+    }
+
+    _getBellConfiguration() {
+        return this.allScm(scm => scm.getBellConfiguration());
+    }
+
+    _getPrInfo(config) {
+        return this.chooseScm(config, scm => scm.getPrInfo(config));
+    }
+
+    _getScmContexts() {
+        return Promise.resolve(Object.keys(this.scms));
+    }
+
+    _canHandleWebhook(headers, payload) {
+        return this.chooseWebhookScm(headers, payload,
+            scm => scm.canHandleWebhook(headers, payload));
+    }
+
+    _getDisplayName(config) {
```

Comment:
`_getDisplayName`  is not needed. We need `getDisplayName` to override `getDisplayName` of scm-base.